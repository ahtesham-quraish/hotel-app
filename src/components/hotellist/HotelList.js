import React, { Component } from 'react';
import './index.css';
import Hotel from '../hotel/Hotel';

export default class HotelList extends Component {
    
  render() {
    const {hotelList} =  this.props; 
    const list = hotelList.map(hotel => <Hotel key={hotel.city} hotel={hotel}/>); 
    return (
        <ul className="card-list">
            {list}
       </ul>
    );
  }
}