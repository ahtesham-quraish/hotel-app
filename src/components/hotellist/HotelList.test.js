import React from 'react';
import ReactDOM from 'react-dom';
import HotelList from './HotelList';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import fetchHotelList from '../../../__mocks__/fetchHotels';
Enzyme.configure({ adapter: new Adapter() });

it('toMatchSnapshot Check', async () => {
  const hotelList = await fetchHotelList();
  const wrapper = shallow(<HotelList hotelList={hotelList.hotels} />);
  expect(wrapper).toMatchSnapshot();
});
test('Check state of the component is mounted ', async () => {
  const hotelList = await fetchHotelList();
  const wrapper = mount(<HotelList hotelList={hotelList.hotels} />);
  expect(wrapper.prop('hotelList').length).toEqual(hotelList.hotels.length);
});
test('Check Hotel Count', async () => {
  const hotelList = await fetchHotelList();
  const wrapper = mount(<HotelList hotelList={hotelList.hotels} />);
  expect(wrapper.find('li').length).toEqual(hotelList.hotels.length);
});
test('Check Class exist or not', async () => {
  const hotelList = await fetchHotelList();
  const wrapper = shallow(<HotelList hotelList={hotelList.hotels} />);
  expect(wrapper.hasClass('card-list')).toEqual(true);
});
