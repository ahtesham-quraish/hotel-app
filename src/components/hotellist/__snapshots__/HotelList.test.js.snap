// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatchSnapshot Check 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <pure(HotelList)
    hotelList={
        Array [
            Object {
              "availability": Array [
                Object {
                  "from": "10-10-2020",
                  "to": "15-10-2020",
                },
                Object {
                  "from": "25-10-2020",
                  "to": "15-11-2020",
                },
                Object {
                  "from": "10-12-2020",
                  "to": "15-12-2020",
                },
              ],
              "city": "dubai",
              "name": "Media One Hotel",
              "price": 102.2,
            },
            Object {
              "availability": Array [
                Object {
                  "from": "10-10-2020",
                  "to": "12-10-2020",
                },
                Object {
                  "from": "25-10-2020",
                  "to": "10-11-2020",
                },
                Object {
                  "from": "05-12-2020",
                  "to": "18-12-2020",
                },
              ],
              "city": "cairo",
              "name": "Rotana Hotel",
              "price": 80.6,
            },
            Object {
              "availability": Array [
                Object {
                  "from": "01-10-2020",
                  "to": "12-10-2020",
                },
                Object {
                  "from": "05-10-2020",
                  "to": "10-11-2020",
                },
                Object {
                  "from": "05-12-2020",
                  "to": "28-12-2020",
                },
              ],
              "city": "london",
              "name": "Le Meridien",
              "price": 89.6,
            },
            Object {
              "availability": Array [
                Object {
                  "from": "04-10-2020",
                  "to": "17-10-2020",
                },
                Object {
                  "from": "16-10-2020",
                  "to": "11-11-2020",
                },
                Object {
                  "from": "01-12-2020",
                  "to": "09-12-2020",
                },
              ],
              "city": "paris",
              "name": "Golden Tulip",
              "price": 109.6,
            },
            Object {
              "availability": Array [
                Object {
                  "from": "20-10-2020",
                  "to": "28-10-2020",
                },
                Object {
                  "from": "04-11-2020",
                  "to": "20-11-2020",
                },
                Object {
                  "from": "08-12-2020",
                  "to": "24-12-2020",
                },
              ],
              "city": "Vienna",
              "name": "Novotel Hotel",
              "price": 111,
            },
            Object {
              "availability": Array [
                Object {
                  "from": "10-10-2020",
                  "to": "19-10-2020",
                },
                Object {
                  "from": "22-10-2020",
                  "to": "22-11-2020",
                },
                Object {
                  "from": "03-12-2020",
                  "to": "20-12-2020",
                },
              ],
              "city": "Manila",
              "name": "Concorde Hotel",
              "price": 79.4,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "hotelList": Array [
        Object {
          "availability": Array [
            Object {
              "from": "10-10-2020",
              "to": "15-10-2020",
            },
            Object {
              "from": "25-10-2020",
              "to": "15-11-2020",
            },
            Object {
              "from": "10-12-2020",
              "to": "15-12-2020",
            },
          ],
          "city": "dubai",
          "name": "Media One Hotel",
          "price": 102.2,
        },
        Object {
          "availability": Array [
            Object {
              "from": "10-10-2020",
              "to": "12-10-2020",
            },
            Object {
              "from": "25-10-2020",
              "to": "10-11-2020",
            },
            Object {
              "from": "05-12-2020",
              "to": "18-12-2020",
            },
          ],
          "city": "cairo",
          "name": "Rotana Hotel",
          "price": 80.6,
        },
        Object {
          "availability": Array [
            Object {
              "from": "01-10-2020",
              "to": "12-10-2020",
            },
            Object {
              "from": "05-10-2020",
              "to": "10-11-2020",
            },
            Object {
              "from": "05-12-2020",
              "to": "28-12-2020",
            },
          ],
          "city": "london",
          "name": "Le Meridien",
          "price": 89.6,
        },
        Object {
          "availability": Array [
            Object {
              "from": "04-10-2020",
              "to": "17-10-2020",
            },
            Object {
              "from": "16-10-2020",
              "to": "11-11-2020",
            },
            Object {
              "from": "01-12-2020",
              "to": "09-12-2020",
            },
          ],
          "city": "paris",
          "name": "Golden Tulip",
          "price": 109.6,
        },
        Object {
          "availability": Array [
            Object {
              "from": "20-10-2020",
              "to": "28-10-2020",
            },
            Object {
              "from": "04-11-2020",
              "to": "20-11-2020",
            },
            Object {
              "from": "08-12-2020",
              "to": "24-12-2020",
            },
          ],
          "city": "Vienna",
          "name": "Novotel Hotel",
          "price": 111,
        },
        Object {
          "availability": Array [
            Object {
              "from": "10-10-2020",
              "to": "19-10-2020",
            },
            Object {
              "from": "22-10-2020",
              "to": "22-11-2020",
            },
            Object {
              "from": "03-12-2020",
              "to": "20-12-2020",
            },
          ],
          "city": "Manila",
          "name": "Concorde Hotel",
          "price": 79.4,
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "hotelList": Array [
          Object {
            "availability": Array [
              Object {
                "from": "10-10-2020",
                "to": "15-10-2020",
              },
              Object {
                "from": "25-10-2020",
                "to": "15-11-2020",
              },
              Object {
                "from": "10-12-2020",
                "to": "15-12-2020",
              },
            ],
            "city": "dubai",
            "name": "Media One Hotel",
            "price": 102.2,
          },
          Object {
            "availability": Array [
              Object {
                "from": "10-10-2020",
                "to": "12-10-2020",
              },
              Object {
                "from": "25-10-2020",
                "to": "10-11-2020",
              },
              Object {
                "from": "05-12-2020",
                "to": "18-12-2020",
              },
            ],
            "city": "cairo",
            "name": "Rotana Hotel",
            "price": 80.6,
          },
          Object {
            "availability": Array [
              Object {
                "from": "01-10-2020",
                "to": "12-10-2020",
              },
              Object {
                "from": "05-10-2020",
                "to": "10-11-2020",
              },
              Object {
                "from": "05-12-2020",
                "to": "28-12-2020",
              },
            ],
            "city": "london",
            "name": "Le Meridien",
            "price": 89.6,
          },
          Object {
            "availability": Array [
              Object {
                "from": "04-10-2020",
                "to": "17-10-2020",
              },
              Object {
                "from": "16-10-2020",
                "to": "11-11-2020",
              },
              Object {
                "from": "01-12-2020",
                "to": "09-12-2020",
              },
            ],
            "city": "paris",
            "name": "Golden Tulip",
            "price": 109.6,
          },
          Object {
            "availability": Array [
              Object {
                "from": "20-10-2020",
                "to": "28-10-2020",
              },
              Object {
                "from": "04-11-2020",
                "to": "20-11-2020",
              },
              Object {
                "from": "08-12-2020",
                "to": "24-12-2020",
              },
            ],
            "city": "Vienna",
            "name": "Novotel Hotel",
            "price": 111,
          },
          Object {
            "availability": Array [
              Object {
                "from": "10-10-2020",
                "to": "19-10-2020",
              },
              Object {
                "from": "22-10-2020",
                "to": "22-11-2020",
              },
              Object {
                "from": "03-12-2020",
                "to": "20-12-2020",
              },
            ],
            "city": "Manila",
            "name": "Concorde Hotel",
            "price": 79.4,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
